core/frontend/src/components/parameter-editor/ParameterEditor.vue

```diff
@@ -0,0 +1,205 @@
+        </v-card-title>
+
+        <v-card-text>
+          <v-container
+            v-if="editedParam"
+          >
+            <v-row>
+              {{ editedParam.description }}
+            </v-row>/
+            <v-row>
+              <v-col
+                cols="6"
+                sm="6"
+                md="6"
+              >
+                <v-select
+                  v-if="editedParam?.options"
+                  v-model.number="editedParam.value"
+                  :items="asComboBoxItems"
+                  @change="saveParam"
+                />
+                <v-checkbox
+                  v-for="(key, value) in editedParam?.bitmask"
+                  v-else-if="editedParam.bitmask"
+                  :key="value"
+                  v-model="editedBitmask"
+                  dense
+                  :label="key"
+                  :value="2 ** value"
+                />
+                <v-text-field
+                  v-else-if="editedParam"
+                  v-model.number="editedParam.value"
+                  label="Value"
+                  type="number"
+                  @change="saveParam"
+                />
+              </v-col>
+            </v-row>
+          </v-container>
+        </v-card-text>
+
+        <v-card-actions>
+          <v-spacer />
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="editDialog = false"
+          >
+            Cancel
+          </v-btn>
+          <v-btn
+            color="blue darken-1"
+            text
+            @click="saveParam"
+          >
+            Save
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+// import SpinningLogo from '../common/SpinningLogo.vue'
+import mavlink2rest from '@/libs/MAVLink2Rest'
+import autopilot_data from '@/store/autopilot'
+import Parameter from '@/types/autopilot/parameter'
+
+export default Vue.extend({
+  name: 'ParameterEditor',
+  data() {
+    return {
+      search: '',
+      editedParam: null as (null | Parameter),
+      editDialog: false,
+      editedBitmask: [] as number[],
+    }
+  },
+  computed: {
+    params() {
+      return autopilot_data.parameters
+    },
+    asComboBoxItems() {
+      const entries = Object.entries(this.editedParam?.options ?? [])
+      return entries.map(([value, name]) => ({ text: name, value: parseInt(value, 10), disabled: false }))
+    },
+  },
+  methods: {
+    editParam(param: Parameter) {
+      if (param.readonly) {
+        return
+      }
+      this.editedParam = param
+      this.editDialog = true
+      this.editedBitmask = []
+      if (param.bitmask === undefined) {
+        console.log('no bitmask!')
+        return
+      }
+      let i = 0
+      for (i = 0; i < 32; i += 1) {
+        // eslint-disable-next-line no-bitwise
+        if (2 ** i & param.value) {
+          this.editedBitmask.push(2 ** i)
+        }
+      }
```

Comment:
You can define variables inside for loops in javascript, like `for (int i = 0;..)`, but the code will be more clear working with bitshift directly.
```suggestion
      for (let v = 1; v < param.value; v = v << 1) {
        // eslint-disable-next-line no-bitwise
        if (v & param.value) {
          this.editedBitmask.push(v)
        }
      }
```